src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+
+[Show me the code](https://github.com/dbanksdesign/style-dictionary-dark-mode/tree/single-token)
+
+1. Get the single-token branch of the demo: `git fetch && git checkout single-token`
+1. Run Style Dictionary: `npm run build`
+1. You should see Style Dictionary generate a bunch of files and then you can run the iOS, Android, and web demos by following the README instructions or the [instructions above](#Project-setup)
+1. You can now toggle light/dark mode in each platform and see the demo app update ✨
+
+Instead of having a separate set of dark token files, we could include a separate dark value in the token itself. This approach more resembles how iOS organizes different color modes inside the color or image asset rather than having separate folders with all colors or images of a single mode. Here is what a token would look like in this approach:
+
+```json
+// tokens/color/background.json5
+{
+  "color": {
+    "background": {
+      "primary": {
+        "value": "{color.core.neutral.0.value}",
+        "darkValue": "{color.core.neutral.1000.value}"
+      }
+    }
+  }
+}
+```
+
+Here we are colocating the default value and a value for dark mode. The name `darkValue` is arbitrary, but we will need to reference it in custom formats and actions later. If you choose to use a different name like nightValue (based on what Android calls it), make sure to update the code later. 
+
+One downside to this approach is that `darkValue` **needs to be a reference** because of how Style Dictionary works. Style Dictionary only transforms the `value` property of a token. Style Dictionary also transforms values *before* references are resolved and references can occur anywhere in Style Dictionary. In the above example `color.core.neutral.1000.value` will be transformed and then `color.background.primary.darkValue` will be resolved to the already transformed value. In practice dark mode tokens having to be a reference to another token is probably ok if you are using a multi-tiered system. 
+
+In this approach we only need to run Style Dictionary once because we have all the color modes in the tokens themselves. The custom formats and actions will then handle using the right value (`.value` or `.darkValue`). This approach needs much heavier customization in custom formats and actions in order to get the right references and values because it is using non-standard data (`.darkValue`). 
+
+But what about component-level tokens that reference this semantic layer of tokens? We don't want to have to duplicate component tokens just to reference `.darkValue` instead of `.value`. To solve this problem, we will use the `outputReference` feature.
+
+### Web
+
+For the web we can't use the built-in `css/variables` format because we need to be able to access `.darkValue` for some tokens. Instead, we can create a wrapper function that wraps a built-in transform and change's any token's `.value` with `.darkValue` if it has one. This feels a bit hacky, but it works. 
```

Comment:
a wrapper function that wraps a built-in transform and change's any token's > a wrapper function that wraps a built-in transform and changes any token's
(this sentence is really hard to follow)